contracts/src/v0.8/functions/dev/v1_X/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -197,4 +201,13 @@ contract TermsOfServiceAllowList is ITermsOfServiceAllowList, IAccessController,
 
     return blockedSenders;
   }
+
+  function updateFromPrevious() external override {
+    address[] memory allowedSenders = ITermsOfServiceAllowList(s_previousToSContract).getAllAllowedSenders();
```

Comment:
Edge case: how will this work when getAllAllowedSenders becomes too big?

Reply:
I was thinking to simply not worry about that at this time given we have <10k users on each network anyway. If we foresee ourselves having to do an upgrade in the future (which who knows if that will happen), I was thinking we could consider this then.
WDYT?

Reply:
The v1.0.0 allowlist contract doesn't allow fetching in batches anyway. It only has the `getAllAllowedSenders()` method which does not allow for fetching in smaller batches.